README.md

```diff
@@ -35,54 +35,107 @@ pod 'Argo', :git => 'https://github.com/thoughtbot/Argo'
 
 Then run `pod install` with CocoaPods 0.36 or newer.
 
-### Git Submodules ###
+### Git Submodules
 
 I guess you could do it this way if that's your thing.
 
 Add this repo as a submodule, and add the project file to your workspace. You
 can then link against `Argo.framework` for your application target.
 
-## Usage
-
-First, create your model. We like to use structs but a class is OK too.
+## Usage tl;dr:
 
 ```swift
 struct User {
   let id: Int
   let name: String
   let email: String?
+  let role: Role
+  let friends: [User]
 }
-```
 
-Then, extend the model to conform to `JSONDecodable`. You will also need to
-write a static constructor method that is curried. We like to use `create`.
-
-```swift
 extension User: JSONDecodable {
-  static func create(id: Int)(name: String)(email: String?) -> User {
-    return User(id: id, name: name, email: email)
+  static func create(id: Int)(name: String)(email: String?)(role: Role)(friends: [User]) -> User {
+    return User(id: id, name: name, email: email, role: role, friends: friends)
+  }
+
+  static func decode(j: JSONValue) -> User? {
+    return User.create
+      <^> j <| "id"
+      <*> j <| "name"
+      <*> j <|? "email" // Use ? for parsing optional values
+      <*> j <| ["role", "title"] // parse nested attributes
+      <*> j <|| "friends" // parse arrays of objects
+  }
+}
+
+// Wherever you receive JSON data:
+
+let json: AnyObject? = NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions(0), error: .None)
```

Comment:
We had discussed wrapping up the `NSData -> JSONValue` part into something. I think we'd still want to keep the `JSONValue -> DecodedType?` part separate though.